crates/plugin_resolve/src/resolver.rs

```diff
@@ -376,6 +376,12 @@ impl Resolver {
             from_str(package_json_info.raw()).unwrap();
 
           for main_field in &self.config.main_fields {
```

Comment:
FWIW, I tried changing `mainFields` config in https://github.com/farm-fe/farm/blob/main/packages/core/src/config/index.ts#L240 which seems like another approach that would be valid. Are main fields hard coded in rs?

Reply:
The default mainFields are `exports`, `browser`, `module`, `main`, see https://github.com/farm-fe/farm/blob/main/crates/core/src/config/mod.rs#L135.

Reply:
> FWIW, I tried changing `mainFields` config in https://github.com/farm-fe/farm/blob/main/packages/core/src/config/index.ts#L240 which seems like another approach that would be valid. Are main fields hard coded in rs?

It is not hard coded, If you configured the mainFields, the default value will be disabled. And there are many methods to determine the package entry, mainFields is just one of them.

Reply:
Hmm, I did some investigation and yes mainField is respected for the entry field, but not for import maps. See this example:
```
{
  "name": "some-package",
  "version": "1.0.0",
  "main": "./main.js",
  "custom": {
    "./implementation.js": "./custom.js"
  }
}
```
It will resolve to `implementation.js` even if `mainFields` have `custom` as first value.

And this:
```
{
  "name": "some-package",
  "version": "1.0.0",
  "main": "./main.js",
  "browser": {
    "./implementation.js": "./browser.js"
  }
}
```
Will resolve to `browser.js` regardless if `browser` is part of `mainFields` or not.

Reply:
IMHO it's probably better approach to use a different default `mainFields` config for node targets (if not set) than ignoring `browser` field.

Reply:
I cross checked this with how esbuild and webpack works, and they both actually work consistently with how fast behaves, so I think you can ignore my comments.